src/Adapter/Cart/CartPresenter.php

```diff
@@ -91,11 +91,17 @@ private function presentProduct(array $rawProduct)
         $rawProduct['pack'] = '';
 
         if ($this->includeTaxes()) {
+            $rawProduct['price_amount'] = $rawProduct['price_wt'];
             $rawProduct['price'] = $this->priceFormatter->format($rawProduct['price_wt']);
         } else {
+            $rawProduct['price_amount'] = $rawProduct['price'];
             $rawProduct['price'] = $rawProduct['price_tax_exc'] = $this->priceFormatter->format($rawProduct['price']);
         }
 
+        if ($rawProduct['price_amount'] && $rawProduct['unit_price_ratio'] > 0) {
+            $rawProduct['unit_price'] = $rawProduct['price_amount'] / $rawProduct['unit_price_ratio'];
```

Comment:
We keep the largest precision here, then the ProductPresenter will format this value properly.